dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilder.cpp

```diff
@@ -365,6 +383,7 @@ namespace winrt::Microsoft::Windows::AppNotifications::Builder::implementation
         xmlResult.append(GetText());
         xmlResult.append(m_attributionText);
         xmlResult.append(GetImages());
+        xmlResult.append(GetProgressBars());
```

Comment:
Same comment here. string.append is expensive. I thought that we addressed this issue somewhere earlier in Paul's pr and decided not to do this?

Reply:
We use printf everywhere except in this function if I remember correclty. I can turn the whole thing into a printf, no problem

Reply:
I believe I only use append in BuildNotification since we reserve the max toast payload size. Append should be 0(1) then since resizing is no longer an issue.

Reply:
This is O(n) for n attributes. Using printf would be O(1) right?

Reply:
Done. Also added a check that we don't exceed the maximin size for the payload.

Reply:
I'm pretty sure the difference in performance is negligiible, if any in this case. doing ~15 append isn't realy O(n) since the set of operation is static. It collap[ses to O(1). Doing 15 append() on pre-reserved memory versus calling printf with 15 parameters is going to endup being the same amount of processing. Strings need to be appended in the end.

The sprintf makes this function match the pattern use almost everywhere in the builder and we can conveniently check the size of the string and throw if it exceeds the max payuyload size.

We could have checked and thrown with the previous approach but it would have looked strange to allocate memeory, then check that it hadn't been exceeded.